debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java

```diff
@@ -142,6 +100,63 @@ protected void validateConnection(Map<String, ConfigValue> configValues, Configu
         }
     }
 
+    private static void checkLoginReplicationRoles(PostgresConnection connection) throws SQLException {
+        if (!connection.queryAndMap(
+                "SELECT r.rolcanlogin AS rolcanlogin, r.rolreplication AS rolreplication," +
+                // for AWS the user might not have directly the rolreplication rights, but can be assigned
+                // to one of those role groups: rds_superuser, rdsadmin or rdsrepladmin
+                        " CAST(array_position(ARRAY(SELECT b.rolname" +
+                        " FROM pg_catalog.pg_auth_members m" +
+                        " JOIN pg_catalog.pg_roles b ON (m.roleid = b.oid)" +
+                        " WHERE m.member = r.oid), 'rds_superuser') AS BOOL) IS TRUE AS aws_superuser" +
+                        ", CAST(array_position(ARRAY(SELECT b.rolname" +
+                        " FROM pg_catalog.pg_auth_members m" +
+                        " JOIN pg_catalog.pg_roles b ON (m.roleid = b.oid)" +
+                        " WHERE m.member = r.oid), 'rdsadmin') AS BOOL) IS TRUE AS aws_admin" +
+                        ", CAST(array_position(ARRAY(SELECT b.rolname" +
+                        " FROM pg_catalog.pg_auth_members m" +
+                        " JOIN pg_catalog.pg_roles b ON (m.roleid = b.oid)" +
+                        " WHERE m.member = r.oid), 'rdsrepladmin') AS BOOL) IS TRUE AS aws_repladmin" +
+                        ", CAST(array_position(ARRAY(SELECT b.rolname" +
+                        " FROM pg_catalog.pg_auth_members m" +
+                        " JOIN pg_catalog.pg_roles b ON (m.roleid = b.oid)" +
+                        " WHERE m.member = r.oid), 'rds_replication') AS BOOL) IS TRUE AS aws_replication" +
+                        " FROM pg_roles r WHERE r.rolname = current_user",
+                connection.singleResultMapper(rs -> rs.getBoolean("rolcanlogin")
+                        && (rs.getBoolean("rolreplication")
+                                || rs.getBoolean("aws_superuser")
+                                || rs.getBoolean("aws_admin")
+                                || rs.getBoolean("aws_repladmin")
+                                || rs.getBoolean("aws_replication")),
+                        "Could not fetch roles"))) {
+            final String errorMessage = "Postgres roles LOGIN and REPLICATION are not assigned to user: " + connection.username();
+            LOGGER.error(errorMessage);
+        }
+    }
+
+    private static void checkWalLevel(ConfigValue hostnameValue, PostgresConnection connection, PostgresConnectorConfig config) throws SQLException {
+        // Logical WAL_LEVEL is only necessary for CDC snapshotting
+        if (config.getSnapshotter().shouldStream()) {
```

Comment:
This is the only business logic change - this check will skip for `initial_only` as it's not needed in this case.